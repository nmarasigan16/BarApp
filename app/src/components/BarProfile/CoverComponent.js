import React, { Component } from 'react';
import PropTypes from 'prop-types';
import {
    View,
    TouchableOpacity
} from 'react-native';
import prompt from 'react-native-prompt-android';
import Text from '../general/text/TextComponent';
import Number from '../general/number/NumberComponent';
import { colorScheme } from "../../lib/styles/ColorScheme";


class CoverComponent extends Component {
    constructor(props) {
        super(props);
        this.state = {
            cover: this.props.cover,
        };

        this.handleCoverReport = this.handleCoverReport.bind(this);

    }

    handleCoverReport(e) {
        prompt(
            'Report Cover',
            'Please enter how much cover is here',
            [
                {text: 'Cancel', onPress: this.cancelPrompt, style: 'cancel'},
                {text: 'OK', onPress: this.confirmPrompt},
            ],
            {
                type: 'plain-text',
                cancelable: true,
                defaultValue: '',
                placeholder: 'placeholder'
            }
        );
    }

    cancelPrompt() {
        console.log('Cancelled');
    }

    confirmPrompt(cover) {
        console.log('Confirmed')
    }

    render() {
        return (
            <View style={{
                flexDirection: 'column',
                alignItems: 'center',
                justifyContent: 'center',
                flex: 1
            }}>
                <Number
                    label={'Cover'}
                    number={`$${this.props.cover}`}
                    />
                <TouchableOpacity onPress={this.handleCoverReport}>
                    <Text
                        style={{
                            color: colorScheme.warn,
                            marginTop: 5
                        }}>
                        report
                    </Text>
                </TouchableOpacity>
            </View>
        )
    }
}

export default CoverComponent;
