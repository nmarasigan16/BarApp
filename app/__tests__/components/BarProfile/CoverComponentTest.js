import 'react-native';
import React from 'react';
import { shallow } from 'enzyme';
import CoverComponent from '../../../src/components/BarProfile/CoverComponent';
import * as prompt from 'react-native-prompt-android';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

it('renders correctly', () => {
    const tree = renderer.create(
        <CoverComponent cover={10}/>
    ).toJSON();

    expect(tree).toMatchSnapshot();
});

it('prompts on press', () => {
    prompt.default = jest.fn();

    const wrapper = shallow(
        <CoverComponent cover={10}/>
    );
    const render = wrapper.dive();
    render.find('TouchableOpacity').first().props().onPress();

    expect(prompt.default).toHaveBeenCalled();
});
