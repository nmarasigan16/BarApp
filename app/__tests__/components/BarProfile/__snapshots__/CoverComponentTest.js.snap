// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "flexDirection": "column",
      "justifyContent": "center",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "column",
        "justifyContent": "center",
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "fontFamily": "Courier New",
          },
          Object {
            "color": "silver",
            "fontSize": 30.607554623001164,
            "margin": 10,
            "textAlign": "center",
          },
        ]
      }
    >
      Cover
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "fontFamily": "Courier New",
          },
          Object {
            "color": "silver",
            "fontSize": 45.911331934501746,
            "fontWeight": "bold",
            "textAlign": "center",
          },
        ]
      }
    >
      $10
    </Text>
  </View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    collapsable={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "fontFamily": "Courier New",
          },
          Object {
            "color": "red",
            "marginTop": 5,
          },
        ]
      }
    >
      report
    </Text>
  </View>
</View>
`;
